tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttTestBase.java

```diff
@@ -137,7 +126,9 @@ public void closeConnectionToMqttAdapter(final VertxTestContext ctx) {
             LOGGER.info("connection to MQTT adapter closed");
             context = null;
             mqttClient = null;
-            ctx.completeNow();
+            // cleanup device registry - done after the adapter connection is closed because otherwise
+            // the adapter would close the connection from its end after having received the device deletion notification
+            helper.deleteObjects(ctx);
```

Comment:
I do not understand why this has to be moved here

Reply:
After test execution, the connection to the MQTT adapter shall be disconnected and the created tenants/devices shall be deleted in the device registry. If the device gets deleted first, the mechanism implemented in this PR will cause the MQTT connection to be disconnected from the adapter side.
This is not a problem per se but when looking at the logs/traces of the test execution, it might be a bit confusing that the MQTT connection may be closed from the device *or* MQTT adapter side, depending on the timing.
I therefore prefer an orderly cleanup, where the MQTT connection is disconnected first.

(For the AMQP tests in #3081 such random behaviour had caused issues because no disconnect handler was set on the AMQP connection opened by the test. I've fixed that since.)

Reply:
Ah, now I see.